python/tchannel/tornado/peer.py

```diff
@@ -411,13 +435,28 @@ def __init__(self, peer, service, **kwargs):
         :param service:
             Name of the service being called through this peer. Defaults to
             an empty string.
+        :param arg_scheme
+            arg scheme type
+        :param retry
+            retry type
+        :param parent_tracing
+            tracing span from parent request
         """
         assert peer, "peer must not be None"
         service = service or ''
 
         self.peer = peer
         self.service = service
-        self.parent_tracing = kwargs.get('parent_tracing', None)
+        self.parent_tracing = parent_tracing
+        self.headers = {
+            'as': arg_scheme or ArgSchemeType.DEFAULT,
+            're': retry or RetryType.DEFAULT,
+        }
+
+        # keep all arguments for retry purpose.
+        # not retry if hostport is set.
+        self._hostport = kwargs.get('hostport', None)
+        self._score_threshold = kwargs.get('score_threshold', None)
```

Comment:
I mean, since you're consuming those arguments in this method, you should add them to the parameter list of the method instead of using `kwargs.get` for them.